app/lib/ui/views/active/form_field/shipper_info_form_field.dart

```diff
@@ -1,112 +1,123 @@
-    _departureNameController.dispose();
-    _contactInfoController.dispose();
-    super.dispose();
-  }
-
-  void save() {
+  void _saveAll() {
     widget.onSaved(ShipperInfo(
-        shipperName: _nameController.text,
-        shipperIsAnimalOwner:
-            _isAnimalOwnerController.text == "Yes" ? true : false,
-        departureLocationId: _departureIdController.text,
-        departureLocationName: _departureNameController.text,
-        departureAddress: _addressFormField.getAddress(),
-        shipperContactInfo: _contactInfoController.text));
+        shipperName: _name,
+        shipperIsAnimalOwner: _isAnimalOwner == "Yes" ? true : false,
+        departureLocationId: _departureId,
+        departureLocationName: _departureLocationName,
+        departureAddress: _departureAddress,
+        shipperContactInfo: _contactInfo));
   }
 
   @override
   Widget build(BuildContext context) {
-    return Column(children: [
-      ListTile(
-          title: Text("Name"),
-          subtitle: TextFormField(
-            controller: _nameController,
-          )),
-      ListTile(
-          visualDensity: VisualDensity(horizontal: 0, vertical: -2),
-          title: Text(
-              "The shipper is the owner of the animals loaded in the vehicle?"),
-          subtitle: DropdownButtonFormField(
-            value: _isAnimalOwnerController.text,
-            items: ["Yes", "No"]
-                .map((label) => DropdownMenuItem(
-                      child: Text(label),
-                      value: label,
-                    ))
-                .toList(),
-            onChanged: (String value) => setState(() {
-              _isAnimalOwnerController.text = value;
-            }),
-          )),
-      ListTile(
-          title: Text("Departure Premises Identification number (PID)"),
-          subtitle: TextFormField(
-            controller: _departureIdController,
-          )),
-      ListTile(
-          title: Text("Name"),
-          subtitle: TextFormField(
-            controller: _departureNameController,
-          )),
-      ListTile(title: Text("Address"), subtitle: _addressFormField),
-      ListTile(
-          title: Text("Shipper’s Contact information in case of emergency"),
-          subtitle: TextFormField(
-            controller: _contactInfoController,
-          )),
-      RaisedButton(child: Text("Save"), onPressed: widget.save)
-    ]);
+    return Form(
+      key: widget.formKey,
+      child: Column(children: [
+        StringFormField(
+            initial: _name,
+            title: "Name",
+            onSaved: (String changed) {
+              _name = changed;
+              _saveAll();
+            },
+            onDelete: Optional.empty()),
+        ListTile(
+            visualDensity: VisualDensity(horizontal: 0, vertical: -2),
+            title: Text(
+                "The shipper is the owner of the animals loaded in the vehicle?"),
+            subtitle: DropdownButtonFormField(
```

Comment:
I can't seem to extract this out as the `onChanged` complains the `(String changed) ` isn't a `(dynamic changed)` 🤷 .